scipy/stats/tests/test_distributions.py

```diff
@@ -6091,6 +6091,27 @@ def test_edge_cases(self):
             assert_equal(stats.triang.cdf(1., 1.), 1)
 
 
+class TestMaxwell:
+
+    # reference values were computed with wolfram alpha
+    # erfc(x/sqrt(2)) + sqrt(2/pi) * x * e^(-x^2/2)
+
+    @pytest.mark.parametrize("x, ref",
+                             [(20, 2.2138865931011177e-86),
+                              (0.01, 0.999999734046458435)])
+    def test_sf(self, x, ref):
+        assert_allclose(stats.maxwell.sf(x), ref, rtol=1e-14)
+
+    # reference values were computed with wolfram alpha
+    # sqrt(2) * sqrt(Q^(-1)(3/2, q))
+
+    @pytest.mark.parametrize("q, ref",
+                             [(0.001, 4.033142223656157022),
+                              (0.99999, 0.0335049635129162318)])
+    def test_isf(self, q, ref):
+        assert_allclose(stats.maxwell.isf(q), ref, rtol=5e-12)
```

Comment:
It looks like the reference value, `0.0335049635129162318`, is the truncation of the value computed by Wolfram Alpha with the expression `sqrt(2) * sqrt(Q^(-1)(3/2, 99999/100000))`.  Because of the sensitivity of the calculation near 1, this value is actually not a good reference value for the input `0.99999`.  The 64 bit floating point number represented by `0.99999` (which is the input to the SciPy calculation) is not exactly equal to the true value `99999/100000` (which is the input used by Wolfram Alpha), and that error gets amplified by the inverse survival functional, resulting in the reference value computed with the input `99999/100000` having a relative error on the order of 1.5e-12 when compared to the value computed with `0.99999` (which is probably why you used 5e-12 for `rtol` below).

When generating reference values from Wolfram Alpha, I find it safer to use inputs that can be represented exactly with 64 bit floating point.  For example, instead of `0.99999`, the value `(2**16 - 1)/2**16 = 0.9999847412109375` can be used.  The result computed by Wolfram Alpha for the expression `sqrt(2) * sqrt(Q^(-1)(3/2, (2^16 - 1)/2^16))` is `0.0385743284050380961345332277294728456492620278268902697685617555...`.  Converted to a 64 bit float, this is `0.0385743284050381`.  Computing the result with the implementation in this pull request, we get `0.038574328405038105`, which has an error of just 1 ULP.

The same problem occurs with the input `0.001` above, but the error introduced by the difference between the 64 bit float `0.001` and the true value 1/1000 used by Wolfram Alpha is on the order of the least significant bit of the result, so it doesn't make a difference.

Neither of the inputs for the test of `isf` are very small, and both of them pass with the main development branch.  To really exercise an implementation of `isf`, an input less than 1e-16 should be used.  The input value `2**-55` (approx. 2.8e-17) will allow the result from Wolfram Alpha to be compared to the 64 bit floating point value computed here.  In Wolfram Alpha, the expression `sqrt(2) * sqrt(Q^(-1)(3/2, 2^-55)` gives `8.9556497471948100851688494017359869084216659670708409658489165804...`. Converted to a 64 bit float, this is `8.95564974719481`.  The value computed by `stats.maxwell.isf(2**-55)` is `8.95564974719481`, so we have zero relative error :+1:.

All that is to explain why I suggest the following changes:
```suggestion
                              (0.9999847412109375, 0.0385743284050381),
                              (2**-55, 8.95564974719481)])
    def test_isf(self, q, ref):
        assert_allclose(stats.maxwell.isf(q), ref, rtol=1e-15)
```

Reply:
Thanks for the very detailed explanation!

These kind of posts should serve as references for anyone implementing such functionality. The details of floating point calculation math is almost never taught properly (admittedly irrelevant in most areas of programming).